app.rb

```diff
@@ -48,57 +53,13 @@ def process_input(option)
   end
   # rubocop:enable Metrics/CyclomaticComplexity
 
-  def list_books()
-    puts "\nList of Books: "
-    if @books.length.zero?
-      puts 'There is no book in the list. Please add a book!'
-    else
-      @books.each_with_index do |book, index|
-        puts "#{index + 1}. #{book.title} by #{book.author}"
-      end
-    end
-    puts "\n"
-  end
-
-  def list_people
-    puts "\nList of Persons: "
-    if @persons.length.zero?
-      puts 'There is no one in the list. Please add a person!'
-    else
-      @persons.each_with_index do |person, index|
-        if person.is_a?(Student)
-          puts "[Student] #{index + 1}. ID: #{person.id}, #{person.name}, #{person.age}"
-        else
-          puts "[Teacher] #{index + 1}. ID: #{person.id}, #{person.name}, #{person.age}"
-        end
-      end
-    end
-    puts "\n"
-  end
-
   def create_book
-    print 'Enter a book title: '
-    title = gets.chomp
-    print 'Enter a book author: '
-    author = gets.chomp
-    @books.push(Book.new(title, author))
+    @list_books.push_book BookCreator.new.return_new_book
     puts 'The book is created successfuly'
   end
 
   def create_rental
-    print "\nSelect a book from the following list by number: "
-    list_books
-    book_index = gets.chomp.to_i
-
-    print "\nSelect a person from the following list by number (not id): "
-    list_people
-    person_index = gets.chomp.to_i
-    print 'Enter a date [format yyyy/mm/dd]: '
-    date = gets.chomp
-
-    rental = Rental.new(date, @persons[person_index - 1], @books[book_index - 1])
-    @rentals << rental
-    puts "Rental created successfully\n\n"
+    @rentals << RentalCreator.new(@list_books, @list_persons).return_new_rental
```

Comment:
- [ ] Kindly make sure to fix the error when I press 6- List all rentals for a given number.

<img width="683" alt="Screenshot 2022-08-29 at 8 00 30 PM" src="https://user-images.githubusercontent.com/70269507/187255889-c154d62a-8c2c-4710-9a1e-5627601120b4.png">